### ISSUE TITLE
Add "Z-Offset" filament override

### ISSUE THREAD

**Issue by Arakon** [supermerill/SuperSlicer#2321]
**Is your feature request related to a problem? Please describe.**
Different materials like different Z-Offsets.. i.e. PETG is generally printed with slightly less squish, while PLA likes a bit more.

**Describe the solution you'd like**
A filament setting override that allows to change Z-Offset specific to a filament, to avoid having to change it in the printer setting or having 2 printer profiles for different materials.

**Comment by ManuelMcLure** [comment]
The workaround I use for this is code like this in my filament-specific start gcode for PETG:
```
G0 Z2
G92 Z1.94 ; For an equivalent 0.06mm raise
```

**Comment by sprior** [comment]
I agree with this request - some materials (like the mentioned PETG) like less first layer squish than others.

**Comment by supermerill** [comment]
duplicate of #171 and #602

**Comment by t0m-r1dd1e** [comment]
I'd also love to see this implemented. My use case is that I want more z offset for flexibles.

**Comment by user897943** [comment]
Just googled this and was thinking exactly that - "why isn't this on the filament overrides?"
Seems a bit of wasted space to have an overrides section with just retraction settings... Started using SuperSlicer yesterday. Hopefully see this implemented soon, it's been open since February 2022...

**Comment by zeroemso** [comment]
i would like to see that too =)

**Comment by Piranha1511** [comment]
Same here!

**Comment by matt-rendall** [comment]
FWIW I'm working around this by passing {filament_type} into my print_start macro where I have adjustment values set per filament type, which I then use with SET_GCODE_OFFSET Z_ADJUST at the very end of my start sequence.

**Comment by pimmen85** [comment]
> The workaround I use for this is code like this in my filament-specific start gcode for PETG:
> 
> ```
> G0 Z2
> G92 Z1.94 ; For an equivalent 0.06mm raise
> ```

this is what i have been looking for, just as a HINT, if i have good z-offset right now with PETG, how do i go about making that specific to the Filament G-code?

and vice versa, with a tighter z-offset for PLA?

**Comment by ipatch** [comment]
> > The workaround I use for this is code like this in my filament-specific start gcode for PETG:
> > ```
> > G0 Z2
> > G92 Z1.94 ; For an equivalent 0.06mm raise
> > ```
> 
> this is what i have been looking for, just as a HINT, if i have good z-offset right now with PETG, how do i go about making that specific to the Filament G-code?
> 
> and vice versa, with a tighter z-offset for PLA?

add it to the custom gcode for the specific filament you have setup in superslicer

![image](https://user-images.githubusercontent.com/613805/224763143-4ec1eb5f-9245-4364-9c5f-8c9a2f1b7a47.png)

**Comment by BrockPlaysFortnite** [comment]
> > > The workaround I use for this is code like this in my filament-specific start gcode for PETG:
> > > ```
> > > G0 Z2
> > > G92 Z1.94 ; For an equivalent 0.06mm raise
> > > ```
> > 
> > 
> > this is what i have been looking for, just as a HINT, if i have good z-offset right now with PETG, how do i go about making that specific to the Filament G-code?
> > and vice versa, with a tighter z-offset for PLA?
> 
> add it to the custom gcode for the specific filament you have setup in superslicer
> 
> ![image](https://user-images.githubusercontent.com/613805/224763143-4ec1eb5f-9245-4364-9c5f-8c9a2f1b7a47.png)

Thank you I was also searching for a way of custom Z for PETG
